### DISCUSSION TITLE
Detach Next.js from the core react-email

### DISCUSSION THREAD

**Discussion by anthonyhagi** [resend/react-email#1746] [Ideas]
### Goals

1. Split the web rendering/preview next js app from the core react-email offering

### Background

- This is useful if you don't want and/or need the web preview next.js app 
- Reduces dependencies when using other frameworks (like Remix)

### Proposal

Splitting the web preview into a separate, installable npm package

**Comment by gabrielmfern** [comment]
The current `react-email` package doesn't include anything except for the CLI, meaning you can have it itself as a dev dependency. React Email's main offering itself is on `@react-email/components` — at least currently. But still, I would agree that not having `next` as a dependency would be much more optimal.

The issue with removing `next` as a dependency is that we need it as a dependency to run the internal Next app for the preview server, even if the user itself is not using it on their application. So, since dev dependencies of libraries won't really be installed on development, we can't really move it to a dev dependency as things are.

One possible option is to bundle, in a tree shaking manner, all the used code from `next` in the CLI. Not sure if that would work or if it would be too much finagling. Open to any ideas.

**Comment by pedro757** [comment]
Is it possible to move that to peerDependencies ? 

It's annoying having two next.js versions installed, especially because Next.js installs a bunch of different packages that are not particularly lightweight

@gabrielmfern

**Reply by gabrielmfern** [reply]
Not really, it wouldn't make much sense here since users can go for React Email without using Next.js. We have a few plans to remove this dependency on Next.js for the next major version that is coming very soon.
